### PR TITLE
[Improvement][Alert] HTTP Related Alert plugin add retry logic

### LINKED ISSUE #14971: [Improvement][Alert] Webex and PagerDuty Alert plugin add retry logic
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Now alert plugins just run once, and when occur network issues will not retry, so add retry logic

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
- close #14971

### CODE REVIEW

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-pagerduty/src/main/java/org/apache/dolphinscheduler/plugin/alert/pagerduty/PagerDutySender.java [hunk 1]
```diff
@@ -53,21 +54,28 @@ public AlertResult sendPagerDutyAlter(String title, String content) {
...
         alertResult.setMessage("send pager duty alert fail.");
 
-        try {
-            sendPagerDutyAlterV2(alertResult, title, content);
-        } catch (Exception e) {
-            log.info("send pager duty alert exception : {}", e.getMessage());
+        for (int i = 0; i < AlertConstants.RETRY_TIMES; i++) {
+            try {
+                int statusCode = sendPagerDutyAlterV2(alertResult, title, content);
+                if (statusCode != AlertConstants.DEFAULT_VALUE) {
```
Comment (fuchanghai) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-pagerduty/src/main/java/org/apache/dolphinscheduler/plugin/alert/pagerduty/PagerDutySender.java]: hi @qingwli  when ```statusCode != HttpStatus.SC_OK``` && ```statusCode != AlertConstants.DEFAULT_VALUE```,do we need retry?

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-pagerduty/src/main/java/org/apache/dolphinscheduler/plugin/alert/pagerduty/PagerDutySender.java]: No, if got statusCode not HttpStatus.SC_OK but actually got a code means the request is correct but got the wrong response, it's not network issue, it's config error

**Review by fuchanghai — APPROVED** [review]
LGTM ,if pass CI

**Comment by qingwli** [comment]
PTAL @Gallardot

**Comment by qingwli** [comment]
@SbloodyS PTAL

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-webexteams/src/main/java/org/apache/dolphinscheduler/plugin/alert/webexteams/WebexTeamsSender.java [hunk 2]
```diff
@@ -67,16 +68,28 @@ public AlertResult sendWebexTeamsAlter(AlertData alertData) {
         alertResult.setStatus("false");
         alertResult.setMessage("send webex teams alert fail.");
 
-        try {
-            send(alertResult, alertData);
-        } catch (Exception e) {
-            log.info("send webex teams alert exception : {}", e.getMessage());
+        for (int i = 0; i < AlertConstants.RETRY_TIMES; i++) {
```
Comment (Gallardot) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-webexteams/src/main/java/org/apache/dolphinscheduler/plugin/alert/webexteams/WebexTeamsSender.java]: I'm voting -1. There is a bad code smell with some duplicate code.

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-webexteams/src/main/java/org/apache/dolphinscheduler/plugin/alert/webexteams/WebexTeamsSender.java]: Yes, some good suggestions about this feature? Different plugin maybe have some same logic, but not all plugins has this logic

Comment (Gallardot) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-webexteams/src/main/java/org/apache/dolphinscheduler/plugin/alert/webexteams/WebexTeamsSender.java]: In fact, many Alert plugins construct an HTTP request and send it to the destination.  The differences lie only in the construction of HTTP headers, HTTP body, HTTP method, and HTTP response parsing.  Would it be possible to establish an HTTP client with a built-in retry function, which all Alert plugins could use?

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-webexteams/src/main/java/org/apache/dolphinscheduler/plugin/alert/webexteams/WebexTeamsSender.java]: Good question, I will update this logic

**Comment by qingwli** [comment]
PTAL @Gallardot @fuchanghai

**Review by Gallardot — APPROVED** [review]
LGTM. Good job!

**Comment by qingwli** [comment]
PTAL @SbloodyS @caishunfeng

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-api/src/main/java/org/apache/dolphinscheduler/alert/api/HttpServiceRetryStrategy.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+public class HttpServiceRetryStrategy implements HttpRequestRetryHandler {
+
+    public static final HttpServiceRetryStrategy retryStrategy = new HttpServiceRetryStrategy();
+
+    private static final int RETRY_COUNT = 3;
+
+    private static final long RETRY_INTERVAL_TIME = 2000L;
+
+    @Override
+    public boolean retryRequest(IOException exception, int executionCount, HttpContext httpContext) {
```
Comment (Gallardot) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-api/src/main/java/org/apache/dolphinscheduler/alert/api/HttpServiceRetryStrategy.java:45]: SonarCloud Quality Gate failed, maybe need some UT for this.

**Comment by qingwli** [comment]
@Radeity

**Review by Radeity — APPROVED** [review]

**Pull request merged**